### DISCUSSION TITLE
Subsequent Livewire requests not generating HTTPS links

### DISCUSSION THREAD

**Discussion by MartinHughes-BPC** [livewire/livewire#4816] [1. Help] [answered]
Hi all,

I have a simple component that lets me search for users and displays a list of them. Each item is wrapped in a link with the url generated by Laravel's route() helper.

In production the links that are being generated are http links and not https links as they should be. I've done a test and it appears that when components are first rendered I do get https links, but on subsequent Livewire requests they change to http links.

The server is behind a reverse proxy , so in Laravel I have configured TrustedProxies  and confirmed that forwarding headers are present on the request. However this hasn't made a difference and I'm still getting http links on the subsequent requests.

So now I'm stuck! Can anybody give me any pointers on how to resolve this?

Many thanks,
Martin.

**Comment by silnarm** [comment]
Laravel uses the scheme based on the incoming connection, so while your initial page was requested over https and you get correct behaviour for that, it seems like your assets (including Livewire) might have been requested and served over http (maybe).
You can use `URL::forceScheme()` to force `route()` to write with the scheme you want, probably also a good idea to make sure the server is configured correctly to force https (which this behaviour suggests is not the case).
Disclaimer: not an ops person.

**Reply by MartinHughes-BPC** [reply]
Thanks for that. I put URL::forceScheme in my AppServiceProvider and that did the trick.

I'm still not sure why it was reverting to http - as far as I could see all assets etc were being served as https. That said all of the infrastructure was set up by the boys in IT and I'm not entirely sure how it all works. It seems that the SSL aspect of the network is actually managed by the proxy (or indeed something else) and when it does its magic it forwards the request on to apache on port 80 or something like that.

Anyway all seems to be working now, so thanks again!

**Comment by MartinHughes-BPC** [comment, marked as answer]
Ok, I did a lot more digging on this. After the change mentioned with forceScheme I found that my email verification links were erroring with an invalid signature.

It turns out that with TrustedProxies you have to ensure your reverse proxy (or whatever) sends the X-Forwarded headers that Laravel/Symfony expect. Out of the box these are:

X-Forwarded-For
X-Forwarded-Host
X-Forwarded-Proto
X-Forwarded-Port 

In my case the first two were being set and not the last two. Making these two headers available in the request made the Trusted Proxy stuff work as it should and I could also remove the call to forceScheme from my application.

All is now working a-ok. I'll leave this here just in case it helps somebody in the future.
